src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -9475,54 +9532,59 @@ namespace SurfaceGeometry {
             auto &vertices = surface.Vertex;
             auto &nSides = surface.Sides;
 
-            bool poppedVertex = true;
-            while (poppedVertex) {
-                poppedVertex = false;
-                Perimeter = 0.0;
+            while (true) {
+                PopCoincidentVertexReturn const popResult = checkPopCoincidentVertex(vertices);
```

Comment:
I hate to see infinite looping structures in the code but I think it is ok here.